src/Particle/LongRange/StructFact.cpp

```diff
@@ -24,7 +24,7 @@ namespace qmcplusplus
 {
 //Constructor - pass arguments to KLists' constructor
 StructFact::StructFact(ParticleSet& P, RealType kc)
-    : DoUpdate(false), StorePerParticle(false), SuperCellEnum(SUPERCELL_BULK)
+    : DoUpdate(false), StorePerParticle(true), SuperCellEnum(SUPERCELL_BULK)
```

Comment:
@ye-luo What do you recommend to do with this?  Enabling this storage is necessary for streaming single particle energy information as needed for energy density and EKT.

Reply:
Call ParticleSet::turnOnPerParticleSK() by the feature related consumer class constructor?

Reply:
I will give it a try.

Reply:
This just results in a segfault as if `StorePerParticle` is `false`.  Same result if called at beginning of each evaluate.

Reply:
Perhaps the resize is wrong?

Reply:
In `StructFact::resize`: `eikr_r.resize(nptcl, nkpts);`

Reply:
In `StructFact::turnOnStorePerParticle`: `eikr_r.resize(nptcl, KLists.numk);`

Reply:
Apparently not the problem.  Suggestions?

Reply:
build debug
gdb your_unit_test

Reply:
Thanks.  I know exactly what is causing the segfault.

Reply:
The underlying issue (again) relates to state machines and order of operations, specifically the order of calls to `StructFact::resize` and `StructFact::turnOnStorePerParticle`.

Reply:
Found the issue.  See latest commit.